### PR TITLE
enable prompt and dscover apis for PebbloSafeRetriever.

### PR DESCRIPTION
**Pull request by rahul-trip**
This PR includes code changes for:

- /app/discover and /prompt apis for  PebbloSafeRetriever apps.
- accept api_key for Pebblo-cloud 

Excludes:

- linting and formatting
- beautification.

### CODE REVIEW

File: libs/langchain/langchain/chains/pebblo_retrieval/base.py [hunk 1]
```diff
@@ -76,6 +104,29 @@ def _call(
...
+                for doc in docs if isinstance(doc, Document)
+                ],
+            "prompt": {
+                "data": question
+            },
+            "response": {
+                "data": answer,
+            },
+            "prompt_time": prompt_time,
+            "user": auth_context.username if auth_context else "unknown",
```
Comment (shreyas-damle) [inline libs/langchain/langchain/chains/pebblo_retrieval/base.py:126]: Do we want to capture it as `unknown` or just keep empty?

**Comment by shreyas-damle** [comment]
Added one minor comment, rest LGTM.

File: libs/langchain/langchain/chains/pebblo_retrieval/models.py [hunk 2]
```diff
@@ -44,6 +43,81 @@ def __init__(self, **data: Any) -> None:
...
+    name: Optional[str] = None
+    version: Optional[str] = None
+    location: Optional[str] = None
+    embedding_model: Optional[str] = None
+
+
+class Chains(BaseModel):
+    name: str
+    model: Optional[Model]
+    vector_dbs: List[VectorDB] | None
```
Comment (Raj725) [inline libs/langchain/langchain/chains/pebblo_retrieval/models.py:92]: This is throwing error in python 3.9

File: libs/langchain/langchain/chains/pebblo_retrieval/base.py [hunk 3]
```diff
@@ -211,3 +284,171 @@ async def _aget_docs(
...
+            {
+                "name": cls.__name__,
+                "model": {
+                    "name": llm.model_name,
+                    "vendor": llm.__class__.__name__
+                    },
+                "vector_dbs": [
+                    {
+                        "name": kwargs['retriever'].vectorstore.__class__.__name__,
+                        "embedding_model": kwargs['retriever'].vectorstore._embeddings.model
```
Comment (Raj725) [inline libs/langchain/langchain/chains/pebblo_retrieval/base.py:449]: Attribute `_embeddings` is not available on all the vectorstores, it's failing for Pinecone

**Review by Raj725 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**